packages/flutter_test/lib/src/widget_tester.dart

```diff
@@ -604,34 +603,11 @@ class WidgetTester extends WidgetController implements HitTestDispatcher, Ticker
     }
   }
 
-  /// Repeatedly calls [pump] with the given `duration` until there are no
-  /// longer any frames scheduled. This will call [pump] at least once, even if
-  /// no frames are scheduled when the function is called, to flush any pending
-  /// microtasks which may themselves schedule a frame.
-  ///
-  /// This essentially waits for all animations to have completed.
-  ///
-  /// If it takes longer that the given `timeout` to settle, then the test will
-  /// fail (this method will throw an exception). In particular, this means that
-  /// if there is an infinite animation in progress (for example, if there is an
-  /// indeterminate progress indicator spinning), this method will throw.
-  ///
-  /// The default timeout is ten minutes, which is longer than most reasonable
-  /// finite animations would last.
-  ///
-  /// If the function returns, it returns the number of pumps that it performed.
-  ///
-  /// In general, it is better practice to figure out exactly why each frame is
-  /// needed, and then to [pump] exactly as many frames as necessary. This will
-  /// help catch regressions where, for instance, an animation is being started
-  /// one frame later than it should.
-  ///
-  /// Alternatively, one can check that the return value from this function
-  /// matches the expected number of pumps.
+  @override
   Future<int> pumpAndSettle([
     Duration duration = const Duration(milliseconds: 100),
-    EnginePhase phase = EnginePhase.sendSemanticsUpdate,
     Duration timeout = const Duration(minutes: 10),
+    EnginePhase phase = EnginePhase.sendSemanticsUpdate,
```

Comment:
One possible way to avoid this (tiny) breaking change is adding a new API `Future<void> settleAnimations()` to `WidgetController`. In it, `LiveWidgetController` can just wait for `!binding.hasScheduledFrame` without `pump` while `WidgetTester` simply calls `pumpAndSettle`. How do you think about that? We could also discuss it with Dan or Ian and see what they prefer.

Reply:
It sounds a good work around. The bad part is for `WidgetController` these two methods sound duplicate. I'm not sure which is better and I agree to discuss further.

Reply:
is it a breaking change?

Reply:
It doesn't break our current test cases because the optional argument is not used in the flutter/flutter repo. @Hixie